### ISSUE TITLE
pyfocal crashed when trying to fit a Gaussian

### ISSUE THREAD

**Issue by hcferguson** [nmearl/pyfocal#65]
Tried to fit a Gaussian to a section of example_spec_ascii. It let me create a model, but crashed when I tried to fit. Traceback is here:
```
(pyf_env)Henrys-MacBook-Pro:specview hcferguson$ pyfocal
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/astropy/units/quantity.py:822: RuntimeWarning: invalid value encountered in true_divide
  return super(Quantity, self).__truediv__(other)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:59: RuntimeWarning: Mean of empty slice.
  warnings.warn("Mean of empty slice.", RuntimeWarning)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:70: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:82: RuntimeWarning: Degrees of freedom <= 0 for slice
  warnings.warn("Degrees of freedom <= 0 for slice", RuntimeWarning)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:94: RuntimeWarning: invalid value encountered in true_divide
  arrmean, rcount, out=arrmean, casting='unsafe', subok=False)
/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/numpy/core/_methods.py:116: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
void QCocoaMenu::insertNative(QCocoaMenuItem *, QCocoaMenuItem *) Menu item is already in a menu, remove it from the other menu first before inserting
Traceback (most recent call last):
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/ui/controller.py", line 135, in fit_model
    fitted_model = apply_model(model, dispersion, flux, fitter_name=fitter_name)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/pyfocal/pyfocal/analysis/modeling.py", line 13, in apply_model
    result = fitter(model, x, y_init)
  File "/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/astropy/modeling/fitting.py", line 462, in __call__
    xtol=acc, full_output=True)
  File "/Users/hcferguson/anaconda/envs/pyf_env/lib/python3.5/site-packages/scipy/optimize/minpack.py", line 380, in leastsq
    raise TypeError('Improper input: N=%s must not exceed M=%s' % (n, m))
TypeError: Improper input: N=3 must not exceed M=0
Abort trap: 6
```

**Comment by nmearl** [comment]
@ibusko Have you come across this error?

**Comment by hcferguson** [comment]
Probably has to do with not bothering to give it anything sensible for an initial guess. When I give it a reasonable initial guess it works.

We might want to think about using some simple statistics to default to sensible initial guesses for some of the standard models.

**Comment by pllim** [comment]
I encountered the same error too. For a given ROI, Pyfocal might be able to update some (if not all) of the initial guesses. For example, central dispersion value of the ROI box can be the new Gaussian mean, and so on.

**Comment by ibusko** [comment]
@nmearl : no, I never got that error. I noticed that when starting with initial guesses that are way off, the fit just won't result in any change of the initial guesses, but a different diagnostic message is produced by the fitter. That's, among other reasons, is that I think the user should somehow be able to read that message.

As for starting the fit with sensible values for the parameters, I found early in the Java specview development that it's definitely a must-have feature. These non-linear fitters can be very finicky and easily get stuck in secondary minima. @pllim suggestion for the starting position of a narrow feature is exactly what that app does by default.

**Comment by nmearl** [comment]
I do not think this is error is because of initial values. I looked through the code and it seems that the mask was being generated from the model layer and not the model layer's parent. In many cases, this meant that there was no data within the ROI area, and thus the error.

I've fixed the issue with the latest PR #72.
